core/domain/html_validation_service_test.py

```diff
@@ -1512,7 +1512,7 @@ def test_svg_string_validation(self):
             ' -595 -23 -112 -14 -299 21 -402 70 -206 212 -368 402 -460 118 -56'
             ' 181 -69 339 -69 104 0 156 4 204 18 255 71 451 258 537 511 35 103'
             ' 44 290 21 402 -60 278 -258 497 -532 586 -88 28 -324 33 -420 9z"/'
-            '> </g> </svg>')
+            ' danger="h4cK3D!"> </g> </svg>')
         self.assertEqual(
-            html_validation_service.get_invalid_svg_tags(invalid_svg_string),
-            set(['testtag']))
+            html_validation_service.get_invalid_svg_tags_and_attrs(
+                invalid_svg_string), (['testtag'], ['danger']))
```

Comment:
Tests are weak. Cover different cases of invalidity -- invalid attr but not tag, invalid tag but not attr, tag masquerading as attr, using an attr that's invalid for the given tag but valid for another tag, etc.